### PR TITLE
Add cleanup to cluster delete

### PR DESCRIPTION
**Pull request by dramich**
Problem: 
The finalizer `controller.cattle.io/cluster-agent-controller` is added to all clusters in v2.0.8 and prior. When upgrading from v2.0.8 to a newer version that lifecycle no longer exists and is left on the controller which stops k8s from deleting the resource. 

Solution:
Add a cleanup to finalizers on cluster delete to ensure that finalizer has been removed from the cluster.

### CODE REVIEW

File: pkg/api/controllers/usercontrollers/usercontroller.go [hunk 1]
```diff
@@ -67,12 +69,19 @@ type userControllersController struct {
...
 	manager       *clustermanager.Manager
 	clusterLister v3.ClusterLister
+	clusters      v3.ClusterInterface
 	ctx           context.Context
 	peers         tpeermanager.Peers
 	start         time.Time
 }
 
 func (u *userControllersController) sync(key string, cluster *v3.Cluster) error {
+	if cluster != nil && cluster.ObjectMeta.DeletionTimestamp != nil {
```
Comment (StrongMonkey) [inline pkg/api/controllers/usercontrollers/usercontroller.go]: nit: `cluster.DeletionTimestamp` should be enough to access.

**Comment by StrongMonkey** [comment]
LGTM

File: pkg/api/controllers/usercontrollers/usercontroller.go (hunk 1 above)
Comment (dramich) [reply pkg/api/controllers/usercontrollers/usercontroller.go]: Good call, updated.

**Comment by cjellick** [comment]
https://github.com/rancher/rancher/issues/15871

**Pull request merged**